artemis-server/src/main/java/org/apache/activemq/artemis/core/persistence/impl/journal/codec/PageCountPendingImpl.java

```diff
@@ -62,21 +67,41 @@ public long getPageID() {
       return pageID;
    }
 
+   /**
+    * @return the persistentSize
+    */
+   @Override
+   public long getPersistentSize() {
+      return persistentSize;
+   }
+
+   /**
+    * @param persistentSize the persistentSize to set
+    */
+   public void setPersistentSize(long persistentSize) {
+      this.persistentSize = persistentSize;
+   }
+
    @Override
    public int getEncodeSize() {
-      return DataConstants.SIZE_LONG * 2;
+      return DataConstants.SIZE_LONG * 3;
    }
 
    @Override
    public void encode(ActiveMQBuffer buffer) {
       buffer.writeLong(queueID);
       buffer.writeLong(pageID);
+      buffer.writeLong(persistentSize);
```

Comment:
@clebertsuconic - here is an example of where the size is now persisted

Reply:
OIC.. you already implemented it..


I would like to have a compatibility test validating that.. there's a journal test on the new compatibility testsuite already... but not one with paging.. and validating the counters...

If you have a way to do it.. it would be great.. otherwise let me know and I will do it.

Reply:
I can add a compatibility test. The value will just be a default (@michaelandrepearce wants me to use -1 instead of 0 and i guess we just would ignore it if not set for the count).

Where's the journal test located in the test suite?

Reply:
JournalCompatibilityTest under compatibility-tests.


I don't think the test is using paging. we should just extend the test to use paging.


I can help you with that if you have questions.